ddtrace/contrib/aioredis/patch.py

```diff
@@ -146,6 +147,8 @@ def _finish_span(future):
             span.finish()
 
     task = func(*args, **kwargs)
+    if not callable(getattr(task, "add_done_callback", None)):
```

Comment:
It might be better to instead do:

```python
if not asyncio.isfuture(task):
    task = asyncio.ensure_future(task)
```

Reply:
A comment here on why/when we get a coroutine vs a future would be great

Reply:
I got this `callable` call idea from other testing cases in this project, `asyncio.isfuture` looks more clear. Changed.
Also added a comment with the reference to the source code where it says when coroutine is returned.